src/include/daos/rpc.h

```diff
@@ -40,23 +40,42 @@
 	 (rpc_ver & RPC_VERSION_MASK) << RPC_VERSION_OFFSET |	\
 	 (mod_id & MODID_MASK) << MODID_OFFSET)
 
+
+#define DAOS_MODULE_LIST \
+	X(DAOS_VOS_MODULE		, 0) /** version object store */ 	\
+	X(DAOS_MGMT_MODULE		, 1) /** storage management */		\
+	X(DAOS_POOL_MODULE		, 2) /** pool service */		\
+	X(DAOS_CONT_MODULE		, 3) /** container service */		\
+	X(DAOS_OBJ_MODULE		, 4) /** object service */		\
+	X(DAOS_REBUILD_MODULE		, 5) /** rebuild **/			\
+	X(DAOS_RSVC_MODULE		, 6) /** replicated service server */	\
+	X(DAOS_RDB_MODULE		, 7) /** rdb */				\
+	X(DAOS_RDBT_MODULE		, 8) /** rdb test */			\
+	X(DAOS_SEC_MODULE		, 9) /** security framework */		\
+	X(DAOS_DTX_MODULE		, 10) /** DTX */			\
+	X(DAOS_PIPELINE_MODULE		, 11)					\
+	X(DAOS_NR_MODULE		, 12) /** number of defined modules */	\
+	X(DAOS_MAX_MODULE		, 64)  /** Size of uint64_t see dmg profile */
+
 enum daos_module_id {
-	DAOS_VOS_MODULE		= 0, /** version object store */
-	DAOS_MGMT_MODULE	= 1, /** storage management */
-	DAOS_POOL_MODULE	= 2, /** pool service */
-	DAOS_CONT_MODULE	= 3, /** container service */
-	DAOS_OBJ_MODULE		= 4, /** object service */
-	DAOS_REBUILD_MODULE	= 5, /** rebuild **/
-	DAOS_RSVC_MODULE	= 6, /** replicated service server */
-	DAOS_RDB_MODULE		= 7, /** rdb */
-	DAOS_RDBT_MODULE	= 8, /** rdb test */
-	DAOS_SEC_MODULE		= 9, /** security framework */
-	DAOS_DTX_MODULE		= 10, /** DTX */
-	DAOS_PIPELINE_MODULE	= 11,
-	DAOS_NR_MODULE		= 12, /** number of defined modules */
-	DAOS_MAX_MODULE		= 64  /** Size of uint64_t see dmg profile */
+#define X(a,b) a = b,
+	DAOS_MODULE_LIST
+#undef X
+
 };
 
+static inline char* daos_opc_to_module_str(uint32_t daos_opc)
+{
+#define X(a,b) case a: return #a;
```

Comment:
You can use variadic macros here, they won't help make this PR smaller but it's better practice and means any future changes to the definitions will not require updates to existing users.
```suggestion
#define X(a, ...) case a: return #a;
```

I'm only commenting here but this applies across this PR obviously.